### ISSUE TITLE
Disable `Naming/AccessorMethodName` on `private` / `protected` methods

### ISSUE THREAD

**Issue by DannyBen** [rubocop/rubocop#11991]
The `Naming/AccessorMethodName` cop complains about methods named `set_*` even when they are `protected` or `private`.

Fixing this violation will result in the code having to use `self.method_name = ...`, which is much worse than having a method start with `set`.

--------

## Expected behavior

The reproduction Ruby code should not throw a `Naming/AccessorMethodName` violation.

## Actual behavior

The reproduction Ruby code throws a `Naming/AccessorMethodName` violation.

## Steps to reproduce the problem

```ruby
# test.rb
# frozen_string_literal: true

# Obligatory comment
class Tester
  def test
    set_vars :production      # good, but in violation
    self.vars = :production   # bad, but passing rubocop
  end

  private

  # Naming/AccessorMethodName violation
  def set_vars(vars)
    p "1: received #{vars}"
  end

  # No violation, but not idiomatic considering `self` must be used when calling
  def vars=(vars)
    p "2: received #{vars}"
  end
end

Tester.new.test
```

```
$ rubocop
```

## RuboCop version

```
$ rubocop -V
1.53.1 (using Parser 3.2.2.3, rubocop-ast 1.28.0, running on ruby 3.1.3) [x86_64-linux]
```

**Comment by bbatsov** [comment]
Reopened — still reproduces on master (1.88). `Naming/AccessorMethodName` flags `def set_vars` even under `private`; it doesn't take method visibility into account.

**Comment by bbatsov** [comment]
Thanks for the writeup, but I'm going to pass on this one. I think `self.foo = ...` reads perfectly fine, and `Naming/AccessorMethodName` is intentionally about consistent naming at every level - a `get_`/`set_` accessor is the same smell whether the method is public or private, so I'd rather not carve out an exception based on visibility. You can always disable the cop locally if it doesn't suit a particular project. Closing as won't fix.
